### PR TITLE
Unix: improve SocketError.SocketError Exception Message

### PR DESCRIPTION
**Pull request by tmds**
This introduces a fake errno for SocketError.SocketError, which
maps to an error message of "Unknown socket error".

### CODE REVIEW

**Comment by tmds** [comment]
@wfurt @stephentoub this is an attempt for fixing https://github.com/dotnet/runtime/issues/35955.
The handling is similar to the existing `EHOSTNOTFOUND` which is also a 'fake' errno.

Another way to fix the issue without fake errno values would be to map the `SocketError` in `SocketException` to a `NativeErrorCode`, and `Message`. Currently the `Message` handling is deferred to the `Win32Exception` base class, which relies on the errno value.

File: src/libraries/System.Net.Primitives/src/System/Net/SocketException.Unix.cs [hunk 1]
```diff
@@ -26,15 +26,12 @@ private SocketException(Interop.ErrorInfo error) : this(SocketErrorPal.GetSocket
...
-                Interop.Error interopErr;
 
-                // If an interop error was not found, then don't invoke Info().RawErrno as that will fail with assert.
-                if (SocketErrorPal.TryGetNativeErrorForSocketError(error, out interopErr))
-                {
-                    nativeErr = interopErr.Info().RawErrno;
-                }
+            Interop.Error interopErr;
+            // If an interop error was not found, then don't invoke Info().RawErrno as that will fail with assert.
+            if (SocketErrorPal.TryGetNativeErrorForSocketError(error, out interopErr))
```
Comment (stephentoub) [inline src/libraries/System.Net.Primitives/src/System/Net/SocketException.Unix.cs:32]:
```suggestion
            // If an interop error was not found, then don't invoke Info().RawErrno as that will fail with assert.
            if (SocketErrorPal.TryGetNativeErrorForSocketError(error, out Interop.Error interopErr))
```

**Review by stephentoub — APPROVED** [review]
Nice. Thanks.

**Review by wfurt — APPROVED** [review]
LGTM

**Pull request merged**